hangman/hangman.py

```diff
@@ -1,27 +1,45 @@
 import random
-my_list = ["I", "eat", "chicken", "and", "turkey"]
-random_word = str(random.choice(my_list).upper())
-print(random_word)
+def get_target_word():
+    my_list = ["I", "eat", "chicken", "and", "turkey"]
+    random_word = str(random.choice(my_list).upper())
+    print(random_word)
+    return random_word
+random_word = get_target_word()
+# To use the variable outside the function
 
-dic_list = {}
-for letters in random_word:
-    dic_list[letters] = False
-print(dic_list)
+def convert_to_dictionary():
+    # this function helps determine which letter 
+    # from the target word is guessed correctly. 
+    dic_list = {}
+    for letters in random_word:
+        dic_list[letters] = False
+    print(dic_list)
+    return dic_list    
+dic_list = convert_to_dictionary()
+# To use the variable outside the function
 
-word = ""
-count = 0
-while True:
-    word = input("Kindly type one letter only: ").upper()
-    index = 0
-    count += 1
-    for i in random_word:
-        if word == i:
-            dic_list[word] = True
-            print("YES in {0} and it only took you {1} tries".format(index, count))
-        index += 1 
-    if word == "!":
-        for val in dic_list:
-            print(val + ": {} ".format(dic_list[val]))
-        break
-    elif word not in dic_list:
-        print("NO! try again. Type ! to quit")
\ No newline at end of file
+def hangman_function():
```

Comment:
I asked for a function for **each turn** that accepts thee word as input. This isn't it.

Reply:
I don't understand

Reply:
this is what you said "A function to process each turn, where all the guessing logic is carried out: This takes in a string containing the input provided by the user and **performs the operations that processes the input**. Nothing is returned by the function."

Reply:
I think he means something like this

```suggestion
def hangman_function(word="TEST"):
```

Reply:
This can also just be named 
```suggestion
def hangman(name="TEST"):
```

Reply:
Yeah, similar but the function would not include the loop since it will be for each turn